icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -4758,12 +4761,12 @@ void RBBITest::TestReverse(std::unique_ptr<RuleBasedBreakIterator>bi) {
     int32_t categoryCount = data->fHeader->fCatCount;
     UCPTrie *trie = data->fTrie;
     bool use8BitsTrie = ucptrie_getValueWidth(trie) == UCPTRIE_VALUE_BITS_8;
-    uint32_t dict_bit = use8BitsTrie ? 0x0080 : 0x4000;
+    uint32_t dictBit = use8BitsTrie ? 0x0080 : 0x4000;
```

Comment:
The history show the value 0x4000 was introduced in Jun 25, 2002 by andy as the landing of https://github.com/unicode-org/icu/commit/32c09250b77da4d691795ac9835279f1d2503f8e#diff-61fbce0535a891d3b5f7aed59ef8598b of ICU-45 new builder for RBBI rules, initial checkin

There are no comment to explain why he use 0x4000 but notice the type of category is  int16_t   not uint16_t. The type of category is changed from int16_t to uint16_t by andy in 2012 as part of https://github.com/unicode-org/icu/commit/5b3a708915b98152de705ae4de25ff000dc60acf

So maybe that is why Andy was using 0x4000 instead of 0x8000 in 2002 to avoid issue with sign bit in int16_t ? I really want to avoid mix unnecessary changes together. If you strongly feel we should try to use 0x8000 instead of keep using 0x4000 , I am more than happy to make such change either BEFORE landing this PR or in a follow up PR but I rather not mix such changes together here.

Reply:
I don't recall the details for choosing for 0x4000 instead of 0x8000. I had thought that it was inherited from the ancient code, but I just went back and checked, and the details were different. The corresponding logic used to be in i18n/dbbi_tbl.cpp. Or maybe I just didn't find the end of the line of the older code.

Frank's guess is probably correct - there were plenty of bits, so avoid any possible sign bit confusion by not using it. And avoid having to double check any unsigned vs signed differences between C++ and Java.

My inclination is to do as little as possible here, in this PR. And follow up with another PR to implement Mark's suggestion of not using a bit at all, but moving all dictionary character categories to the end, and keeping track of the boundary between dictionary and non-dictionary categories.

This would be consistent between 8 and 16 bit categories, make the most efficient use of the available 8 bit range, and avoid a bunch of ugly conditionals for choosing the dictionary bit. It also needs some further looking into the code that assigns the category numbers.

Reply:
noted for follow-up